318 harden time extraction

ndaelman-hu opened the pull request:
Closes #318

ndaelman-hu committed:
Capture malformed time representations in `totalsc` that cannot be deconsctructed as 2 floating point numbers

Changes to be committed:
	modified:   electronicparsers/vasp/parser.py

ndaelman-hu committed:
Capture malformed time representations in `total` that cannot be deconstructed as 2 floating point numbers

Changes to be committed:
	modified:   electronicparsers/vasp/parser.py

ndaelman-hu committed:
Apply ruff

ndaelman-hu commented on electronicparsers/vasp/parser.py:
```diff
@@ -1494,14 +1494,19 @@ def get_tier(self, type='native') -> Union[str, None]:
+        time = [
+            t if isinstance(t, list) and len(t) == 2 else [None, None] for t in time
+        ]
         if time and len(np.shape(time)) != 2:
```
@copilot you mean that numpy won't handle mixed datatypes well?

ndaelman-hu committed:
Apply reviewer feedback:
- Handle mixed datatypes in numpy array

Changes to be committed:
	modified:   electronicparsers/vasp/parser.py

ndaelman-hu committed:
Fix failing pytests:
- Disambiguate between cpu and wall time
- Add catches for each combination

Changes to be committed:
	modified:   electronicparsers/vasp/parser.py

ladinesa commented on electronicparsers/vasp/parser.py:
```diff
@@ -2285,10 +2323,30 @@ def parse_dos(n_calc):
+                    elif len(time) > 0 and time[0] is not None:
+                        sec_scc.time_calculation = float(time[0])
+                        sec_scc.time_physical = time_initial + sec_scc.time_calculation
+                elif time is not None:
```
you already have this in l2330

ndaelman-hu replied:
I centralized the logic more

ladinesa reviewed (commented):
Please add test

ndaelman-hu committed:
Add test case for malformed time tag in VASP

Changes to be committed:
	new file:   tests/data/vasp/malformed_time/vasprun.xml
	modified:   tests/test_vaspparser.py

ndaelman-hu committed:
Apply reviewer comment

Changes to be committed:
	modified:   electronicparsers/vasp/parser.py
